apps/examples/security_test/security_test_main.c

```diff
@@ -0,0 +1,1597 @@
+					   0x61, 0xab, 0x7f, 0x6e, 0x17, 0x04, 0x35, 0x1a, 0x19, 0xd0, 0xb5, 0x37, 0xfe, 0xc7, 0xbd,
+					   0xc4, 0xc4, 0x48, 0xca, 0xfa, 0x96, 0xfe, 0x86, 0x25, 0x81, 0xd0, 0xb0, 0x76, 0x04, 0xb8,
+					   0x94, 0x37, 0x6c, 0x17, 0x46, 0xc7, 0x04, 0xc9, 0xd8, 0x92, 0x38, 0x41, 0x32, 0x87, 0x63,
+					   0x97, 0x22, 0x2e, 0x5f, 0x0a, 0x0b, 0x2f, 0xa1, 0x55, 0x95, 0x67, 0xb1, 0x9b, 0xbe, 0xb1,
+					   0x06, 0x8a, 0x1c, 0x10, 0xa8, 0x10, 0x46, 0xe7, 0xe7, 0x14, 0xd2, 0x3c, 0xf9, 0xb5, 0xff,
+					   0xf3, 0x36, 0xda, 0x62, 0xce, 0xeb, 0x36, 0xbe, 0x3d, 0xce, 0x6c, 0x99, 0xbd, 0xe1, 0x14,
+					   0x29, 0x51, 0x4f, 0x5b, 0x64, 0xe4, 0x34, 0x6b, 0xad, 0x74, 0xfe, 0x5d, 0xc0, 0x6f, 0x4f,
+					   0xb7, 0x6a, 0x1e, 0x4c, 0xc2, 0xdb, 0x73, 0x11, 0x09, 0x91, 0x77, 0x2f, 0xfc, 0x8e, 0x8a,
+					   0xd7, 0xf5, 0x42, 0x7a, 0x2c, 0x61, 0x7f, 0x0f, 0xb0, 0xe3, 0x32, 0x99, 0x54, 0xc8, 0x26,
+					   0xbf, 0xbe, 0xf0, 0xe9, 0x0a, 0x8e, 0xd4, 0x59, 0x85, 0x60, 0x8d, 0x1b, 0x32, 0x6b, 0xe0,
+					   0x6e
+				    };
+
+	unsigned char output[300];
+	unsigned int object_id[2] = { OID_DH_1024, OID_DH_2048 }, olen;
+
+	printf("----------------------------------------------\n");
+	printf("TESTCASE : see_generate/compute_dhm_params\n");
+	printf("----------------------------------------------\n");
+
+	memset(&d_param, 0, sizeof(struct sDH_PARAM));
+
+	d_param.modules_p = p_buf_1024;
+	d_param.generator_g = g_buf_1024;
+	d_param.modules_p_byte_len = 128;
+	d_param.generator_g_byte_len = 128;
+
+	d_param.publickey = malloc(d_param.modules_p_byte_len);
+	d_param.publickey_byte_len = d_param.modules_p_byte_len;
+
+	memset(d_param.publickey, 0, d_param.publickey_byte_len);
+	d_param.object_id = object_id[0];
+
+	for (i = 0; i < 9; i++) {
+		see_selfprintf("[%d] ", i);
+
+		ret = see_generate_dhm_params(&d_param, i);
+		ret += see_compute_dhm_param(&d_param, i, output, &olen);
+		if (i == 8 && ret) {
+			see_selfprintf(" success\n");
+		} else if (ret) {
+			test_result |= 0x200;
+			see_selfprintf(" fail\n");
+		} else {
+			see_selfprintf(" success\n");
+		}
+	}
+
+	free(d_param.publickey);
+
+	d_param.modules_p = p_buf_2048;
+	d_param.generator_g = g_buf_2048;
+	d_param.modules_p_byte_len = 256;
+	d_param.generator_g_byte_len = 256;
+
+	d_param.publickey = malloc(d_param.modules_p_byte_len);
+	d_param.publickey_byte_len = d_param.modules_p_byte_len;
+
+	memset(d_param.publickey, 0, d_param.publickey_byte_len);
+	d_param.object_id = object_id[1];
+
+	for (i = 0; i < 9; i++) {
+		see_selfprintf("[%d] ", i);
+
+		ret = see_generate_dhm_params(&d_param, i);
+		ret += see_compute_dhm_param(&d_param, i, output, &olen);
+		if (i == 8 && ret) {
+			see_selfprintf(" success\n");
+		} else if (ret) {
+			test_result |= 0x200;
+			see_selfprintf(" fail\n");
+		} else {
+			see_selfprintf(" success\n");
+		}
+	}
+
+	free(d_param.publickey);
+
+	isp_clear(0);
+	return test_result;
+}
+
+unsigned int TC_see_rsa_encrypt_decrypt(void)
```

Comment:
Is this tc? then, why don't you put this into testcase example?

Reply:
Yes, this is the initial version. So this may be put into the testcase example later..